evm/p2p/les.py

```diff
@@ -343,3 +355,54 @@ def send_get_contract_code(self, block_hash: bytes, key: bytes, request_id: int)
         }
         header, body = GetContractCodes(self.cmd_id_offset).encode(data)
         self.send(header, body)
+
+
+class StatusV2(Status):
+    _cmd_id = 0
+
+    def __init__(self, id_offset: int) -> None:
+        super().__init__(id_offset)
+        self.items_sedes['announceType'] = sedes.big_endian_int
+
+
+class GetProofsV2(GetProofs):
+    _cmd_id = 15
+
+
+class ProofsV2(Command):
+    _cmd_id = 16
+    structure = [
+        ('request_id', sedes.big_endian_int),
+        ('buffer_value', sedes.big_endian_int),
+        ('proof', sedes.CountableList(sedes.raw)),
+    ]
+
+
+class LESProtocolV2(LESProtocol):
+    version = 2
+    _commands = [StatusV2, Announce, BlockHeaders, BlockBodies, Receipts, ProofsV2, ContractCodes]
+    handshake_msg_type = StatusV2
+    cmd_length = 21
+
+    def send_handshake(self, head_info):
+        resp = {
+            'announceType': LES_ANNOUNCE_SIMPLE,
+            'protocolVersion': self.version,
+            'networkId': self.peer.network_id,
+            'headTd': head_info.total_difficulty,
+            'headHash': head_info.block_hash,
+            'headNum': head_info.block_number,
+            'genesisHash': head_info.genesis_hash,
+        }
+        cmd = StatusV2(self.cmd_id_offset)
+        self.send(*cmd.encode(resp))
+        self.logger.debug("Sending LES/Status msg: %s", resp)
+
+    def send_get_proof(self, block_hash: bytes, account_key: bytes, key: bytes, from_level: int,
```

Comment:
*linting nitpick*

Can we drop all of these to their own line.  I'm finding that signatures with type annotations and more than 2-3 arguments are really hard to read when they are on the same line.